rpc/providers/web-socket-proxy.ts

```diff
@@ -0,0 +1,120 @@
+type ReconnectOptions = {
+  attempts?: number;
+  maxAttempts?: number;
+  delay?: number;
+};
+
+type WebSocketClientOptions = {
+  webSocketFactory: () => WebSocket;
+  reconnect?: ReconnectOptions;
+};
+
+export class WebSocketProxy implements Pick<WebSocket, "send" | "close"> {
```

Comment:
We don't need to utilize `WebSocketProxy` in slots expecting something conforming to the Ws interface. Why the `implements` clause?

Reply:
I don't think the name `WebSocketProxy` is fitting, given that we also use the word proxy to refer to our conduit to the on-chain world. Might I recommend `WsContainer`? Any other ideas?

Reply:
WsContainer sounds good